src/cache/intern.rs

```diff
@@ -31,11 +31,14 @@ cfg_if! {
 
         fn intern_str(key: &str) {
             CACHE.with(|cache| {
-                let mut cache = cache.entries.borrow_mut();
+                let entries = &cache.entries;
 
                 // Can't use `entry` because `entry` requires a `String`
-                if !cache.contains_key(key) {
-                    cache.insert(key.to_owned(), JsValue::from(key));
+                if !entries.borrow().contains_key(key) {
+                    // Note: we must not hold the borrow while we create the `JsValue`,
+                    // because it will try to look up the value in the cache first.
+                    let value = JsValue::from(key);
+                    entries.borrow_mut().insert(key.to_owned(), value);
```

Comment:
Why the change here?

Reply:
As the comment says, because now `JsValue::from` goes via generic conversion which tries to look up the string in cache while we're still holding the `RefMut`.

The other alternative fix is to change lookup code from `cache.borrow()` to `cache.try_borrow()?`, perhaps that's cleaner.

Reply:
Aaah, I misunderstood the comment for just the regular Rust lifetime stuff because we already hold a mutable reference. So I was surprised why it worked before then.

That's fine then.